examples/demo_search/public/demo_search_strategy.ts

```diff
@@ -54,15 +50,16 @@ import { DEMO_SEARCH_STRATEGY, IDemoResponse } from '../common';
  * @param search - a search function to access other strategies that have already been registered.
  */
 export const demoClientSearchStrategyProvider: TSearchStrategyProvider<typeof DEMO_SEARCH_STRATEGY> = (
-  context: ISearchContext,
-  search: ISearchGeneric
+  context: ISearchContext
 ): ISearchStrategy<typeof DEMO_SEARCH_STRATEGY> => {
   return {
-    search: (request, options) =>
-      search(
+    search: (request, options) => {
+      const syncStrategyProvider = context.getSearchStrategy(SYNC_SEARCH_STRATEGY);
+      if (!syncStrategyProvider) throw new Error('Search strategy not found');
+      return syncStrategyProvider(context).search(
         { ...request, serverStrategy: DEMO_SEARCH_STRATEGY },
-        options,
-        SYNC_SEARCH_STRATEGY
-      ) as Observable<IDemoResponse>,
+        options
+      ) as Observable<IDemoResponse>;
```

Comment:
Nit: Can we move the `getSearchStrategy` call out of the `search` function? We don't need to look it up every time a consumer calls `search` (I don't think):

```ts
const syncStrategyProvider = context.getSearchStrategy(SYNC_SEARCH_STRATEGY);
const syncStrategy = syncStrategyProvider(context);
return {
  search: (request, options) => syncStrategy.search({
    ...request,
    serverStrategy: DEMO_SEARCH_STRATEGY
  }) as Observable<IDemoResponse>;
}
```